File: scripts/upload_to_shopify.js
import { readFile } from 'fs/promises';
import { fileURLToPath } from 'url';
import { dirname, resolve } from 'path';
import fetch from 'node-fetch';
import dotenv from 'dotenv';

dotenv.config();

const API_VERSION = '2023-07';

export async function uploadToShopify(product) {
  const res = await fetch(
    `https://${process.env.SHOPIFY_STORE}/admin/api/${API_VERSION}/products.json`,
    {
      method: 'POST',
      headers: {
        'X-Shopify-Access-Token': process.env.SHOPIFY_TOKEN,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({ product }),
    }
  );

  if (!res.ok) {
    const text = await res.text();
    throw new Error(`Shopify API error ${res.status}: ${text}`);
  }

  const data = await res.json();
  console.log(
    data.product
      ? `🛒 Producto subido: ${data.product.title}`
      : `❌ Error: ${JSON.stringify(data.errors)}`
  );
}

if (process.argv[1] === fileURLToPath(import.meta.url)) {
  (async () => {
    try {
      const fileArg = process.argv[2] || 'products/tryon-products.json';
      const filePath = resolve(dirname(fileURLToPath(import.meta.url)), '..', fileArg);
      const content = await readFile(filePath, 'utf-8');
      const items = JSON.parse(content);
      const products = Array.isArray(items) ? items : [items];
      for (const product of products) {
        await uploadToShopify(product);
      }
    } catch (err) {
      console.error(err);
      process.exit(1);
    }
  })();
}

File: scripts/upload_to_shopify.py
import os
import json
import csv
import requests

API_KEY = os.getenv("SHOPIFY_API_KEY")
PASSWORD = os.getenv("SHOPIFY_API_PASSWORD")
SHOP_NAME = os.getenv("SHOPIFY_STORE")
API_VERSION = os.getenv("SHOPIFY_API_VERSION", "2023-04")

if not (API_KEY and PASSWORD and SHOP_NAME):
    raise SystemExit("Missing Shopify credentials")

BASE_URL = f"https://{API_KEY}:{PASSWORD}@{SHOP_NAME}.myshopify.com/admin/api/{API_VERSION}"


def upload_product(product):
    url = f"{BASE_URL}/products.json"
    response = requests.post(url, json={"product": product})
    response.raise_for_status()
    return response.json()


def load_products(path):
    if path.endswith(".json"):
        with open(path) as f:
            return json.load(f)
    elif path.endswith(".csv"):
        with open(path, newline="") as f:
            reader = csv.DictReader(f)
            return [row for row in reader]
    raise ValueError("Unsupported file type")


if __name__ == "__main__":
    file_path = os.getenv("PRODUCT_FILE", "products.json")
    products = load_products(file_path)
    for product in products:
        data = upload_product(product)
        print(f"Uploaded {data['product']['title']}")
